crates/burn-ndarray/src/ops/interpolate.rs

```diff
@@ -0,0 +1,214 @@
+use burn_tensor::ElementConversion;
+use ndarray::Array4;
+#[cfg(not(feature = "std"))]
+use num_traits::Float;
+
+use crate::{iter_range_par, run_par, FloatNdArrayElement, NdArrayTensor, UnsafeSharedRef};
+
+pub(crate) fn nearest_interpolate<E: FloatNdArrayElement>(
+    x: NdArrayTensor<E, 4>,
+    output_size: [usize; 2],
+) -> NdArrayTensor<E, 4> {
+    let x = x.array.into_dimensionality::<ndarray::Ix4>().unwrap();
+
+    let (batch_size, channels, in_height, in_width) = x.dim();
+    let [out_height, out_width] = output_size;
+
+    let y_ratio = (in_height as f64) / (out_height as f64);
+    let x_ratio = (in_width as f64) / (out_width as f64);
+
+    let out_element_num = batch_size * channels * out_height * out_width;
+    let strides = (
+        channels * out_height * out_width,
+        out_height * out_width,
+        out_width,
+    );
+
+    let mut output = Array4::zeros((batch_size, channels, out_height, out_width));
+    let unsafe_shared_out = UnsafeSharedRef::new(&mut output);
+
+    run_par!(|| {
+        iter_range_par!(0, out_element_num).for_each(|id| {
+            let (b, c, h, w) = (
+                id / strides.0,
+                id % strides.0 / strides.1,
+                id % strides.1 / strides.2,
+                id % strides.2,
+            );
+
+            let y_in = (y_ratio * h as f64).floor() as usize;
```

Comment:
I think It's great that we are keeping this consistent with OpenCV version of this operation. Just as FYI, some methods may use ceil() or round() as well to find nearest coordinate so we may see Candle/Tch backend's implementation output slightly differ.  But i think this makes sense.